AwsCryptographicMaterialProviders/dafny/AwsCryptographyKeyStoreAdmin/src/Mutations.dfy

```diff
@@ -87,10 +95,38 @@ module {:options "/functionSyntax:4" } Mutations {
     modifies keyManagerStrategy.Modifies
     ensures keyManagerStrategy.ValidState()
   {
-    var kmsOperation: string;
+
     var success?: bool := false;
     var throwAwayError;
-
+    // TODO-HV-2-M3: Support mutations on HV-2 item (mutation starting with hv-2 item)
+    :- Need(
+      item.EncryptionContext[Structure.HIERARCHY_VERSION] == Structure.HIERARCHY_VERSION_VALUE_1,
+      Types.KeyStoreAdminException(
+        message := "At this time, Mutations ONLY support HV-1; BK's Active Item is HV-2.")
+    );
+    if (isTerminalHv2?) {
+      // TODO-HV-2-M2: Add test to cover the if condition of this code path
+      // TODO-HV-2-M4: Support other key manager strategy
+      :- Need(keyManagerStrategy.kmsSimple?, Types.KeyStoreAdminException(message:="Only KeyManagementStrategy.AwsKmsSimple is allowed when mutating to hv-2."));
```

Comment:
Blocking:
Move this error to the Admin Error Message file:
https://github.com/aws/aws-cryptographic-material-providers-library/blob/hv-2/hv-2/AwsCryptographicMaterialProviders/dafny/AwsCryptographyKeyStoreAdmin/src/KeyStoreAdminErrorMessages.dfy

And then refer to that here and in the test case.

Reply:
Done